include/nuttx/spinlock.h

```diff
@@ -35,6 +35,7 @@
 #ifndef CONFIG_SPINLOCK
 typedef struct
 {
+  uint8_t unused;    /* NULL structure may caused build issue. */
```

Comment:
Why does this even exists? When CONFIG_SPINLOCK not defied then why is it needed? Is the proper fix to ifdef the offending code?

Reply:
This would require many places:
```
ifdef CONFIG_SPINLOCK
struct spinlock_s xxx;
#endif
```
another approach is define the spinlock by macro like:
https://github.com/apache/incubator-nuttx/blob/master/include/nuttx/power/pm.h#L90-L91
https://github.com/apache/incubator-nuttx/blob/master/include/nuttx/power/pm.h#L802

what do you think? @masayuki2009 and @davids5 .

Reply:
@xiaoxiang781216  I would vote for DECLARE_SPINLOCK and avoid the ifdef rash

Reply:
@easonxiang please drop the change in spinlock.h, I will provide a patch to add DECLARE_SPINLOCK macro instead.

Reply:
done.